boards/st/nucleo_g431kb/nucleo_g431kb_defconfig

```diff
@@ -0,0 +1,24 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# serial drivers
+CONFIG_SERIAL=y
+CONFIG_UART_INTERRUPT_DRIVEN=y
+
+# Console
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+# enable GPIO
+CONFIG_GPIO=y
+
+# enable clocks
+CONFIG_CLOCK_CONTROL=y
+
+# Enable MPU
+CONFIG_ARM_MPU=y
+
+# Enable HW stack protection
+CONFIG_HW_STACK_PROTECTION=y
+
+# enable pin controller
+CONFIG_PINCTRL=y
```

Comment:
does not go in boards

Reply:
Where sould this go? I looked up two different boards and they also put it in defconfig: 
boards/st/nucleo_f042k6/nucleo_f042k6_defconfig
boards/st/nucleo_g431rb/nucleo_g431rb_defconfig

Reply:
Drop it entirely

Reply:
How can i drop it entirely? If i omit it the pinctrl driver is not included in the build and i end up with errors during the link.

Reply:
Rebase your PR

Reply:
How would you like it to be reseolved? In every st nucleo board port i have been looking in it was done this way. Am i missing something? If i ommit it from the defconfig the pinctrl- driver gets not pulled in and the link fails with undefined externals.

Reply:
> Rebase your PR

You have not done this

Reply:
I rebased my PR on main and force pused it again but i am not shure if i did it correctly, now i committed all chances from main again in my branch

Reply:
this line is still present and needs removing

Reply:
Would you mind giving me a hint how to remove this without getting undefinded externals due to the missing pinctrl- driver?

Reply:
> Rebase your PR

> this line is still present and needs removing

Reply:
I rebased my PR already, did i miss something? If i remove line 24 from the nucleo_g432kb_defconfig the pinctrl driver is not build and the build for the examples e. h. blinky, hello_world, blinky_pwm fails

Reply:
Thanks a lot for your help!
I did the failing test before rebasing my branch and then did not test again. Now it works perfectly fine without the Configuration setting.